### DISCUSSION TITLE
ZX7D00CE01S Support 7 inch RGB

### DISCUSSION THREAD

**Discussion by cos-it-is** [moononournation/Arduino_GFX#361] [General]
Can you add support for this screen? [Here](http://en.smartpanle.com/product-item-19.html) 

I believe it should be

define TFT_BL 45
Arduino_ESP32RGBPanel* rgbpanel = new Arduino_ESP32RGBPanel(
39 /* DE /, 38 / VSYNC /, 5 / HSYNC /, 9 / PCLK /,
10 / R0 /, 11 / R1 /, 12 / R2 /, 13 / R3 /, 14 / R4 /,
21 / G0 /, 0 / G1 /, 46 / G2 /, 3 / G3 /, 8 / G4 /, 18 / G5 /,
17 / B0 /, 16 / B1 /, 15 / B2 /, 7 / B3 /, 6 / B4 /,
0 / hsync_polarity /, 20 / hsync_front_porch /, 1 / hsync_pulse_width /, 87 / hsync_back_porch /,
0 / vsync_polarity /, 5 / vsync_front_porch /, 1 / vsync_pulse_width /, 31 / vsync_back_porch /,
1 / pclk_active_neg /, 24000000 / prefer_speed /,
0 / de_idle_high /, 0 / pclk_idle_high */);

Arduino_RGB_Display* gfx = new Arduino_RGB_Display(
800 /* width /, 480 / height */, rgbpanel);.

However, I seem to get an initial white flush, washing effect but then black screen.

**Comment by moononournation** [comment]
please also provide a PDQgraphicstest Youtube video link for this display.

**Comment by cos-it-is** [comment]
> please also provide a PDQgraphicstest Youtube video link for this display.

It's not currently working. I had a previous screen and it worked. You can see here:


https://youtu.be/qXRFZEE0E-8?feature=shared

Now I bought a new new screen (should be exactly the same) and same config but this happens:

https://youtu.be/C-H4saEgjLM?feature=shared

I have no idea why. Exactly the same version of hardware.

**Reply by moononournation** [reply]
the maximum speed should be 16MHz only

**Reply by cos-it-is** [reply]
I've tried that also. Same result. This is what I previously used. 24Mhz is what the vendor recommended.

At 16Mhz = same result.

**Reply by cos-it-is** [reply]
I'm really confused by this. I think hardware issue but if I flash original bin from manufacturer. (Lvgl) It works.

**Reply by cos-it-is** [reply]
EK9716BD4 is the screen driver IC

**Reply by holsteinjohan** [reply]
My display is working however it takes 20 seconds to startup, very weird. I suspect the timing.
Here is my working setup for the display, please help 
`// parallel display
Arduino_ESP32RGBPanel *bus = new Arduino_ESP32RGBPanel(
  39 /* DE */, 38 /* VSYNC */, 5 /* HSYNC */, 9 /* PCLK */,
  10 /* R0 */, 11 /* R1 */, 12 /* R2 */, 13 /* R3 */, 14 /* R4 */,
  21  /* G0 */, 0 /* G1 */, 46 /* G2 */, 3 /* G3 */, 8 /* G4 */, 18 /* G5 */,
  17  /* B0 */,16 /* B1 */, 15 /* B2 */, 7 /* B3 */,6 /* B4 */,

  0 /* hsync_polarity */, 40 /* hsync_front_porch */, 10 /* hsync_pulse_width */, 40 /* hsync_back_porch */,
  0 /* vsync_polarity */, 1 /* vsync_front_porch */, 31 /* vsync_pulse_width */, 13 /* vsync_back_porch */,
  1 /* pclk_active_neg */, 24000000 /* prefer_speed */, false /*useBigEndian*/
);

Arduino_RGB_Display *gfx = new Arduino_RGB_Display(
  800 /* width */, 480 /* height */, bus /* *rgbpanel*/);`

**Reply by holsteinjohan** [reply]
The type is : ZX7D00CE01S

**Reply by cos-it-is** [reply]
Try this, it worked well for me on previous screen. I've ordered another and will try on that. Maybe I just had a dodgy board.


Arduino_ESP32RGBPanel* rgbpanel = new Arduino_ESP32RGBPanel(
  39 /* DE */, 38 /* VSYNC */, 5 /* HSYNC */, 9 /* PCLK */,
  10 /* R0 */, 11 /* R1 */, 12 /* R2 */, 13 /* R3 */, 14 /* R4 */,
  21 /* G0 */, 0 /* G1 */, 46 /* G2 */, 3 /* G3 */, 8 /* G4 */, 18 /* G5 */,
  17 /* B0 */, 16 /* B1 */, 15 /* B2 */, 7 /* B3 */, 6 /* B4 */,
  0 /* hsync_polarity */, 20 /* hsync_front_porch */, 1 /* hsync_pulse_width */, 87 /* hsync_back_porch */,
  0 /* vsync_polarity */, 5 /* vsync_front_porch */, 1 /* vsync_pulse_width */, 31 /* vsync_back_porch */,
  1 /* pclk_active_neg */, 16000000 /* prefer_speed */,
  0 /* de_idle_high */, 0 /* pclk_idle_high */);

Arduino_RGB_Display* gfx = new Arduino_RGB_Display(
  800 /* width */, 480 /* height */, rgbpanel);

**Reply by holsteinjohan** [reply]
I changed the code and work awesome.

I also need now the touch screen code..... can someone help me ?
Thank you.
Johan Holstein

**Comment by holsteinjohan** [comment]
`// parallel display
Arduino_ESP32RGBPanel *bus = new Arduino_ESP32RGBPanel(
  39 /* DE */, 38 /* VSYNC */, 5 /* HSYNC */, 9 /* PCLK */,
  10 /* R0 */, 11 /* R1 */, 12 /* R2 */, 13 /* R3 */, 14 /* R4 */,
  21  /* G0 */, 0 /* G1 */, 46 /* G2 */, 3 /* G3 */, 8 /* G4 */, 18 /* G5 */,
  17  /* B0 */,16 /* B1 */, 15 /* B2 */, 7 /* B3 */,6 /* B4 */,

  0 /* hsync_polarity */, 0 /* hsync_front_porch */, 210 /* hsync_pulse_width */, 30 /* hsync_back_porch */,
  0 /* vsync_polarity */, 0 /* vsync_front_porch */, 22 /* vsync_pulse_width */, 13 /* vsync_back_porch */,
  1 /* pclk_active_neg */, 16000000 /* prefer_speed */
);
Arduino_RGB_Display *gfx = new Arduino_RGB_Display(
  800 /* width */, 480 /* height */, bus);
`

**Reply by holsteinjohan** [reply]
I also need now the touch screen code..... can someone help me ?

Thank you.
Johan Holstein

**Reply by holsteinjohan** [reply]
https://www.youtube.com/watch?v=zp3WjRNN8Pk

**Reply by holsteinjohan** [reply]
Touchscreen is working too, case closed.
Thanks for helping!
Best regards,
Johan Holstein

**Reply by cos-it-is** [reply]
Can you share touch screen code?

**Reply by cos-it-is** [reply]
also screen shot of flash settings in arduino?

**Reply by cos-it-is** [reply]
Your settings fixed my new screen too! hoorah!

**Reply by moononournation** [reply]
You may share a PDQgraphicstest Youtube link, if you found it works. Then I add the settings to the example.

**Reply by cos-it-is** [reply]
https://youtu.be/r-zAMUzpkGE

**Comment by moononournation** [comment]
Ok, committed.

**Reply by holsteinjohan** [reply]
Hi everyone,

I minimized the  code but it is  a working sketch including touchscreen
Good luck and have fun!

Johan Holstein

`// J.G. Holstein @2023 test program
// library versions
#include <Arduino_GFX_Library.h> // 1.38
#include "TAMC_GT911.h"


// https://oleddisplay.squix.ch/
// please provide your own fonts 
#include "fonts/DejaVu_Sans_Mono_9.h"   // Default font for virtual displays
#include "fonts/DejaVu_Sans_Mono_64.h"  // Adafruit GFX font
#include "fonts/DejaVu_Sans_Mono_22.h"  // Adafruit GFX font
#include "fonts/DejaVu_Sans_Mono_28.h"  // Adafruit GFX font
#include "fonts/DejaVu_Sans_Mono_32.h"  // Adafruit GFX font
#include "fonts/DejaVu_Sans_Mono_40.h"  // Adafruit GFX font
#include "fonts/DejaVu_Sans_Mono_48.h"  // Adafruit GFX font
#include "fonts/DejaVu_Sans_Mono_96.h"  // Adafruit GFX font

// parallel display
Arduino_ESP32RGBPanel *bus = new Arduino_ESP32RGBPanel(
  39 /* DE */, 38 /* VSYNC */, 5 /* HSYNC */, 9 /* PCLK */,
  10 /* R0 */, 11 /* R1 */, 12 /* R2 */, 13 /* R3 */, 14 /* R4 */,
  21 /* G0 */, 0 /* G1 */, 46 /* G2 */, 3 /* G3 */, 8 /* G4 */, 18 /* G5 */,
  17 /* B0 */, 16 /* B1 */, 15 /* B2 */, 7 /* B3 */, 6 /* B4 */,

  0 /* hsync_polarity */, 0 /* hsync_front_porch */, 210 /* hsync_pulse_width */, 30 /* hsync_back_porch */,
  0 /* vsync_polarity */, 0 /* vsync_front_porch */, 22 /* vsync_pulse_width */, 13 /* vsync_back_porch */,
  1 /* pclk_active_neg */, 12000000 /* prefer_speed */,
  0 /* de_idle_high */, 0 /* pclk_idle_high */
);
Arduino_RGB_Display *gfx = new Arduino_RGB_Display(
  800 /* width */, 481 /* height */, bus);  // yes 481 due problems in display driver! 

// backlite
#define LCD_PIN_BK_LIGHT 45
#define PWM_CHANNEL 0
#define PWM_FREQ 5000  // Hz
#define pwm_resolution_bits 8

// tft intensity range 0 - 255
int dutyCycle = 100;

// touchscreen
#define TOUCH_IIC_SDA 48
#define TOUCH_IIC_SCL 47
#define TOUCH_INT -1
#define TOUCH_RST -1
#define TOUCH_WIDTH 800
#define TOUCH_HEIGHT 480
TAMC_GT911 tp = TAMC_GT911(TOUCH_IIC_SDA, TOUCH_IIC_SCL, TOUCH_INT, TOUCH_RST, TOUCH_WIDTH, TOUCH_HEIGHT);

bool TouchZ;
int  TouchY;
int  TouchX;

void setup() {
  gfx->begin();
  gfx->fillScreen(BLACK);
  Serial.begin(115200);

  ledcSetup(PWM_CHANNEL, PWM_FREQ, pwm_resolution_bits);
  ledcAttachPin(LCD_PIN_BK_LIGHT, PWM_CHANNEL);
  ledcWrite(PWM_CHANNEL, dutyCycle);

  tp.begin();
  tp.setRotation(ROTATION_INVERTED);
  delay(500);

  gfx->setTextColor(WHITE, BLACK);
  gfx->setCursor(300, 200);
  gfx->setFont(&DejaVu_Sans_Mono_32);
  gfx->print("Hi everyone");
}

void loop() {

  tp.read();

  if (tp.isTouched) {
    for (int i = 0; i < tp.touches; i++) {

      Serial.print("  x: ");
      Serial.print(TouchX);
      TouchX = (tp.points[i].x);
      Serial.print("  y: ");
      Serial.print(TouchY);
      TouchY = (tp.points[i].y);
      TouchZ = (tp.isTouched);
      Serial.println(' ');
    }
    tp.isTouched = false;
  }
}
 `

**Reply by holsteinjohan** [reply]
Please add * in the comments like 39 /* DE* /,

**Reply by holsteinjohan** [reply]
Hi Panlee users look what this display can do.... awesome and very cheap too.

https://youtu.be/4HdSYjHcWZI

Regards,
Johan

**Reply by cos-it-is** [reply]
Does anyone have issues with screen flicker when connected to WiFi?

**Reply by moononournation** [reply]
yes, it can be. any long run loop can flick it

**Reply by cos-it-is** [reply]
Is there a way to avoid this? It's really irritating.

Maybe settings need slight adjustment and if so, where to look?

**Comment by holsteinjohan** [comment]
No problem with WiFi however this need a bit tweaking works awesome.JohanVerstuurd vanaf mijn iPhoneOp 21 nov 2023 om 16:51 heeft orange pill ***@***.***> het volgende geschreven:﻿
Is there a way to avoid this? It's really irritating.
Maybe settings need slight adjustment and if so, where to look?

**Reply by cos-it-is** [reply]
No, I mean the wifi works, but when enabled - i get a single pixel line flicker right at the top of the screen. do you get this?

**Reply by TheHexaMaster** [reply]
Same display, same problem with flickering when wifi on. Someone find solution? :)

**Reply by cos-it-is** [reply]
@moononournation this could be a library issue?

**Reply by moononournation** [reply]
if some RGB display works fine but some not. It should be parameters issue. I did not have this display in hand, and cannot help on it.

**Reply by Emile1154** [reply]
on other display ESP32-8048S070 I had a problem with Wi-Fi too, and fix this reducing the frequency PCLK to 12 MHz, in timings configuration: 
Arduino_RPi_DPI_RGBPanel *gfx = new Arduino_RPi_DPI_RGBPanel(
  bus,
  800     /* width */,
    0     /* hsync_polarity */,
    8     /* hsync_front_porch */,
    10     /* hsync_pulse_width */, // old val 2
    43     /* hsync_back_porch */,  // old val 43
  480     /* height */, 
    0     /* vsync_polarity */,
    8     /* vsync_front_porch */,
    8     /* vsync_pulse_width */,  //old 2
    12     /* vsync_back_porch */,   //old 12
    1     /* pclk_active_neg */,
12000000  /* prefer_speed */,  //old 16M 
  true    /* auto_flush */
);

**Reply by Emile1154** [reply]
can be done work this display without additional libraries gfx. Arduino framework have esp-idf library for lcd panels, code init display for LVGL on  ZX7D00CE01S: 

```
static lv_disp_drv_t disp_drv;
static lv_disp_draw_buf_t draw_buf;
  
esp_lcd_panel_handle_t g_panel_handle = NULL;
esp_lcd_rgb_panel_config_t panel_config = {
      .clk_src = LCD_CLK_SRC_PLL240M,
      .timings = {
          .pclk_hz = 16000000,        //16M ?
          .h_res = WIDTH,
          .v_res = HEIGH,
          .hsync_pulse_width  = 210,    //10
          .hsync_back_porch   =  40,     //40
          .hsync_front_porch  =   0,    //40
          .vsync_pulse_width  =  22,     //6
          .vsync_back_porch   =  13,     //31
          .vsync_front_porch  =   0,    //18
          .flags = {
              .hsync_idle_low = true, // hsync falling edge 
              .vsync_idle_low = true, // vsync falling edge
              .de_idle_high = false,   // de stay stay low
              .pclk_active_neg = true, // pclk falling edge
              .pclk_idle_high = false, // pclk stay low
          },
      },
      .data_width = 16,
      .psram_trans_align = 64,
      .hsync_gpio_num = LCD_HSYNC_GPIO,
      .vsync_gpio_num = LCD_VSYNC_GPIO,
      .de_gpio_num = LCD_DE_GPIO,
      .pclk_gpio_num = LCD_PCLK_GPIO,
      .data_gpio_nums = {
          LCD_D0_GPIO,
          LCD_D1_GPIO,
          LCD_D2_GPIO,
          LCD_D3_GPIO,
          LCD_D4_GPIO,
          LCD_D5_GPIO,
          LCD_D6_GPIO,
          LCD_D7_GPIO,
          LCD_D8_GPIO,
          LCD_D9_GPIO,
          LCD_D10_GPIO,
          LCD_D11_GPIO,
          LCD_D12_GPIO,
          LCD_D13_GPIO,
          LCD_D14_GPIO,
          LCD_D15_GPIO,
      },
      .disp_gpio_num = LCD_DISP_EN_GPIO,
      .on_frame_trans_done = example_notify_lvgl_flush_ready,
      .user_ctx = &disp_drv,
      .flags ={ 
          .relax_on_idle = false,   // Mannually control refresh operation
          .fb_in_psram = true,
   },
};
  
ESP_ERROR_CHECK(esp_lcd_new_rgb_panel(&panel_config, &g_panel_handle));

ESP_ERROR_CHECK(esp_lcd_panel_reset(g_panel_handle));
ESP_ERROR_CHECK(esp_lcd_panel_init(g_panel_handle));

lv_init();

//recommended buffer > 1/10 screen size
static lv_color_t *buf1 = (lv_color_t *) heap_caps_malloc(sizeof(lv_color_t)*WIDTH*HEIGH/8,MALLOC_CAP_INTERNAL | MALLOC_CAP_8BIT);
assert(buf1);
static lv_color_t *buf2 = (lv_color_t *) heap_caps_malloc(sizeof(lv_color_t)*WIDTH*HEIGH/8, MALLOC_CAP_INTERNAL | MALLOC_CAP_8BIT);
assert(buf2);
lv_disp_draw_buf_init(&draw_buf, buf1, buf2, WIDTH*HEIGH/8);

lv_disp_drv_init(&disp_drv);
disp_drv.hor_res = WIDTH;
disp_drv.ver_res = HEIGH;
disp_drv.flush_cb = my_disp_flush;
disp_drv.draw_buf = &draw_buf;     
disp_drv.full_refresh = 1; 
disp_drv.user_data = g_panel_handle;
current_display = lv_disp_drv_register(&disp_drv);

ESP_ERROR_CHECK(gpio_set_level((gpio_num_t) LCD_PIN_BK_LIGHT, LCD_BK_LIGHT_ON_LEVEL)); //enable light
```

callback method for link on_frame_trans_done:

```
static bool example_notify_lvgl_flush_ready(esp_lcd_panel_handle_t panel, esp_lcd_rgb_panel_event_data_t *event_data, void *user_data)
{
    lv_disp_drv_t *disp_driver = (lv_disp_drv_t *)user_data;
    lv_disp_flush_ready(disp_driver);
    return false;
}
```
update disp method:
```
void my_disp_flush(lv_disp_drv_t *disp, const lv_area_t *area, lv_color_t *color_p)
{
    esp_lcd_panel_handle_t panel_handle = (esp_lcd_panel_handle_t) disp->user_data;
    int offsetx1 = area->x1;
    int offsetx2 = area->x2;
    int offsety1 = area->y1;
    int offsety2 = area->y2;

    esp_lcd_panel_draw_bitmap(panel_handle, offsetx1, offsety1, offsetx2 + 1, offsety2 + 1, color_p);
    lv_disp_flush_ready(disp);
}
```

`

**Reply by cos-it-is** [reply]
Thanks I've tried this but it does nothing.

It's like a single white line flicker at the top of the screen.

**Reply by Emile1154** [reply]
I have it too, but from bottom. I'm planning to experiment with command `esp_lcd_panel_set_gap(esp_lcd_panel_handle_t panel, int x_gap, int y_gap)` for decide this problem

**Reply by Emile1154** [reply]
problem solved after change HEGHT from 480 to #define HEIGH 481

**Reply by moononournation** [reply]
that mean you should adjust the vsync related parameters.

**Reply by cos-it-is** [reply]
Can confirm changing to 481 is fixed for me also. Thanks!!!

**Comment by holsteinjohan** [comment]
The Panlee display get a keyboard and mouse.... see video.
https://www.youtube.com/watch?v=viNG4bxeqJA

Best.
Johan
